java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java

```diff
@@ -299,50 +299,106 @@
+  // the default extension formatting tDo be " extn: 3456", then you should specify " extn: " here
   // as the default extension prefix. This can be overridden by region-specific preferences.
   private static final String DEFAULT_EXTN_PREFIX = " ext. ";
 
-  // Pattern to capture digits used in an extension. Places a maximum length of "7" for an
-  // extension.
-  private static final String CAPTURING_EXTN_DIGITS = "(" + DIGITS + "{1,7})";
   // Regexp of all possible ways to write extensions, for use when parsing. This will be run as a
   // case-insensitive regexp match. Wide character versions are also provided after each ASCII
   // version.
   private static final String EXTN_PATTERNS_FOR_PARSING;
   static final String EXTN_PATTERNS_FOR_MATCHING;
   static {
-    // One-character symbols that can be used to indicate an extension.
-    String singleExtnSymbolsForMatching = "x\uFF58#\uFF03~\uFF5E";
-    // For parsing, we are slightly more lenient in our interpretation than for matching. Here we
-    // allow "comma" and "semicolon" as possible extension indicators. When matching, these are
-    // hardly ever used to indicate this.
-    String singleExtnSymbolsForParsing = ",;" + singleExtnSymbolsForMatching;
-
-    EXTN_PATTERNS_FOR_PARSING = createExtnPattern(singleExtnSymbolsForParsing);
-    EXTN_PATTERNS_FOR_MATCHING = createExtnPattern(singleExtnSymbolsForMatching);
-  }
-
-  /**
-   * Helper initialiser method to create the regular-expression pattern to match extensions,
-   * allowing the one-char extension symbols provided by {@code singleExtnSymbols}.
-   */
-  private static String createExtnPattern(String singleExtnSymbols) {
-    // There are three regular expressions here. The first covers RFC 3966 format, where the
-    // extension is added using ";ext=". The second more generic one starts with optional white
-    // space and ends with an optional full stop (.), followed by zero or more spaces/tabs/commas
-    // and then the numbers themselves. The other one covers the special case of American numbers
-    // where the extension is written with a hash at the end, such as "- 503#"
-    // Note that the only capturing groups should be around the digits that you want to capture as
-    // part of the extension, or else parsing will fail!
-    // Canonical-equivalence doesn't seem to be an option with Android java, so we allow two options
-    // for representing the accented o - the character itself, and one in the unicode decomposed
-    // form with the combining acute accent.
-    return (RFC3966_EXTN_PREFIX + CAPTURING_EXTN_DIGITS + "|" + "[ \u00A0\\t,]*"
-        + "(?:e?xt(?:ensi(?:o\u0301?|\u00F3))?n?|\uFF45?\uFF58\uFF54\uFF4E?|"
-        + "\u0434\u043E\u0431|" + "[" + singleExtnSymbols + "]|int|anexo|\uFF49\uFF4E\uFF54)"
-        + "[:\\.\uFF0E]?[ \u00A0\\t,-]*" + CAPTURING_EXTN_DIGITS + "#?|"
-        + "[- ]+(" + DIGITS + "{1,5})#");
+    EXTN_PATTERNS_FOR_PARSING = createExtnPattern(true);
```

Comment:
attach to line 309 (see internal CL comment for more details)

Reply:
DOne.